Support environment in Workflow/ProductDefinition Properties

chrisvire opened the pull request:
* Add ProductDefinition Properties
* Merge ProductionDefinition/WorkflowDefinition Properties into ProcessProperties
* Merge ProcessProperties with ActionParameters so that environment can flow to BuildEngine
* explicitly declare environment parameter in default_workflow.sql

Note: Currently, only explicitly defined parameters in the workflow are persisted in WorkflowProcessSchemeInstance. I have contacted DWKit on the correct way to handle this generically.

davidmoore1 commented on source/OptimaJet.DWKit.StarterApplication/OptimaJet.DWKit.StarterApplication.csproj:
```diff
@@ -19,12 +19,6 @@
-    <LaunchBrowser>false</LaunchBrowser>
-    <ApplicationURL>http://localhost:5000</ApplicationURL>
-  </PropertyGroup>
-
```
Don't know about this one one way or the other.  Just wanted to make sure this is intentional

chrisvire commented:
> Just check the one change I added the comment on to make sure it was intentional.

I reverted that change. I don't know why it is there. However, I didn't intentionally change the file.
